pkg/xen-tools/initrd/udhcpc_script.sh

```diff
@@ -35,7 +39,16 @@ case "$1" in
     # configure interface and routes
     ip addr flush dev $interface
     ip addr add ${ip}/${mask} dev $interface
-    [ -n "$router" ] && ip route add default via ${router%% *} dev $interface
+    [ -n "$router" ] && ip route add "${router%% *}" src "$ip" dev "$interface"
+
+    [ -n "$router" ] && ip route add "${router%% *}" src "$ip" dev "$interface" table "$TABLE"
+    [ -n "$router" ] && ip route add default via "${router%% *}" table "$TABLE"
+    [ -n "$router" ] && ip rule add from "${ip}"/32 table "$TABLE"
+    [ -n "$router" ] && ip rule add to "${ip}"/32 table "$TABLE"
```

Comment:
Why do we need a "to" rule? The host would be able to talk to itself without finding a default route in a particular table.